onnxruntime/core/providers/tensorrt/tensorrt_execution_provider.cc

```diff
@@ -1779,11 +1856,29 @@ common::Status TensorrtExecutionProvider::Compile(const std::vector<FusedNodeAnd
           trt_config->setDLACore(trt_state->dla_core);
         }
 
+        // Load timing cache from file. Create a fresh cache if the file doesn't exist
+        std::unique_ptr<nvinfer1::ITimingCache> timing_cache = nullptr;
+        if (trt_state->timing_cache_enable) {
+            std::vector<char> loaded_timing_cache = loadTimingCacheFile(timing_cache_path);
+            timing_cache.reset(trt_config->createTimingCache(static_cast<const void*>(loaded_timing_cache.data()), loaded_timing_cache.size()));
+            if (timing_cache == nullptr) {
+              return ORT_MAKE_STATUS(ONNXRUNTIME, EP_FAIL,
+                                     "TensorRT EP could not create timing cache: " + timing_cache_path);
+            }
+            trt_config->setTimingCache(*timing_cache, force_timing_cache_match_);
+            LOGS_DEFAULT(VERBOSE) << "[TensorRT EP] Deserialized timing cache from " + timing_cache_path;
+        }
+
         // Build engine
         {
           auto lock = GetApiLock();
+          auto engine_build_start = std::chrono::steady_clock::now();
           *(trt_state->engine) = std::unique_ptr<nvinfer1::ICudaEngine>(
               trt_builder->buildEngineWithConfig(*trt_state->network->get(), *trt_config));
+          auto engine_build_stop = std::chrono::steady_clock::now();
+          LOGS_DEFAULT(INFO) << "TensorRT engine build for " << trt_state->trt_node_name_with_precision << " took: " <<
```

Comment:
We need to avoid adding logging code in inference/compute function since it might have impact on performance. Could we add a flag so that user can decide whether to show the engine build time (including calling steady_clock())?

Reply:
Thanks for adding this. If you can fix the indentation that will be great!

Reply:
```
          if (detailed_build_log_) {
            auto engine_build_stop = std::chrono::steady_clock::now();
            LOGS_DEFAULT(INFO) << "TensorRT engine build for " << trt_node_name_with_precision << " took: " <<
              std::chrono::duration_cast<std::chrono::milliseconds>(engine_build_stop - engine_build_start).count() << "ms" << std::endl;
          }

```
Do you mean indent 2 spaces (above) ? Or should i indent until `"TensorRT...` What is the guidance here. Also is there a wait to autoformat with vscode on onnxruntime?

Reply:
the indent 2 spaces. Yes, that's the fix, thanks.

Reply:
yes, VS code should be able to use the https://github.com/microsoft/onnxruntime/blob/main/.clang-format
https://code.visualstudio.com/docs/cpp/cpp-ide#_code-formatting

Reply:
Ok so clang format is used but not enforced on PRs ? Because when i used that it also formatted other parts that i did not touch and didn't want to push format changes.

Reply:
Yes, clang format is not strictly enforced for c++ on PRs currently.